src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +275,27 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this.currentChangeFieldUpdateEvent));
+            this.currentChangeFieldUpdateEvent = new ChangeFieldUpdateEvent();
+            newEntry.registerListener(this.currentChangeFieldUpdateEvent);
         }
+
         entry = Optional.ofNullable(newEntry);
 
         updateLayout();
         update();
     }
 
+    private class ChangeFieldUpdateEvent {
```

Comment:
Sorry, but I still don't get why the private class is necessary. As soon as some entry is changed you call PreviewPanel.update. Why should this call be in a private subclass? Can you please provide an example.